src/nix/flake.cc

```diff
@@ -957,6 +957,7 @@ struct CmdFlakeArchive : FlakeCommand, MixJSON, MixDryRun
 struct CmdFlakeShow : FlakeCommand, MixJSON
 {
     bool showLegacy = false;
+    bool showForeign = false;
```

Comment:
I'm not keen on is having `showForeign` false by default, because it would be bad for backwards-compatibility – esp. for the json output which is designed to be used in scripts and is more dangerous to break.

Reply:
I totally understand your sentiments, though having it defaulting to `true` is a major annoyance for the human.

1. Evaluation of `nix flake show` will take much longer than needed
2. Evaluation of `nix flake show` will show a lot of redundant information
3. Evaluation of `nix flake show` will fail for a majority of flakes

We can avoid all of these 3 problems by defaulting it to `false`.

At the same time, breaking the interface is fine, as long as we do it **now**. After 3.0 has been released we can not do it anymore.

Back when 2.4 got released, no one cared that it broke a lot of scripts. Back then it has been pointed out that its at your own risk to use an experimental API. So… If someone complaints, why did they use an experimental API if they do not want things to change?

Also, it is much easier to add `--foreign` to a handful of scripts, but annoying to have to remember to add it to each and every invocation to shave off **minutes** of eval time and annoying IFD errors…

I'd even suggest to make this a `nix.conf` option, such that every **user** can decide on their very own default in the systems or users `nix.conf`.

Reply:
This should definitely not be a `nix.conf` option, as tempting as that is, since it's a bad idea for scripting to have options that fundamentally alter what the output of a command is. You can end up with a script that unexpectedly fails on another machine or user account because `foreign` is set to a different value...